### ISSUE TITLE
NullPointerException in cvssToSonarQubeSeverity with dependency-check-maven:aggregate

### ISSUE THREAD

**Issue by miksonx** [dependency-check/dependency-check-sonar-plugin#107]
Hi,

We use Bamboo 6.2.3 build 60210 
Sonar plugin for bamboo 1.6.1
org.owasp:dependency-check-maven:aggregate
SonarQube 6.7.6
sonar-dependency-check-plugin-1.1.3.jar (tried 1.1.2 and 1.1.0 - same behavior) 

The bamboo does not complain on sonar plugin incompatibility.
The dependency check report is generated in target folder.
This looks like parsing or sending the report to SQ does not work.
Noticed that Dependency check tab under project in SQ does not provide any information only blank screen.
![image](https://user-images.githubusercontent.com/5308184/52979964-49770480-33d8-11e9-85c8-1d7c898ec2fe.png)

No error in the SQ log is listed.

Here is full stack trace of the error
```
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.400 15/15 source files have been analyzed
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.401 Java Main Files AST scan (done) | time=1400ms
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.401 Java Test Files AST scan
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.402 1 source files to be analyzed
build	18-Feb-2019 17:16:42	[DEBUG] 17:16:42.403 'src/test/java/AppTest.java' generated metadata as test  with charset 'UTF-8'
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.416 Java Test Files AST scan (done) | time=15ms
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.416 Sensor JavaSquidSensor [java] (done) | time=1884ms
uild	18-Feb-2019 17:16:42	[INFO] 17:16:42.417 Sensor Dependency-Check [dependencycheck]
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.417 Process Dependency-Check report
build	18-Feb-2019 17:16:42	[INFO] 17:16:42.422 1/1 source files have been analyzed
build	18-Feb-2019 17:16:42	[INFO] ------------------------------------------------------------------------
build	18-Feb-2019 17:16:42	[INFO] BUILD FAILURE
build	18-Feb-2019 17:16:42	[INFO] ------------------------------------------------------------------------
build	18-Feb-2019 17:16:42	[INFO] Total time: 6.026 s
build	18-Feb-2019 17:16:42	[INFO] Finished at: 2019-02-18T17:16:42+01:00
build	18-Feb-2019 17:16:42	[INFO] Final Memory: 23M/490M
build	18-Feb-2019 17:16:42	[INFO] ------------------------------------------------------------------------
build	18-Feb-2019 17:16:42	[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project xxx: null: MojoExecutionException: NullPointerException -> [Help 1]
build	18-Feb-2019 17:16:42	org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project xxx: null
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
build	18-Feb-2019 17:16:42		at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
build	18-Feb-2019 17:16:42		at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
build	18-Feb-2019 17:16:42		at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
build	18-Feb-2019 17:16:42		at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
build	18-Feb-2019 17:16:42		at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
build	18-Feb-2019 17:16:42		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
build	18-Feb-2019 17:16:42		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
build	18-Feb-2019 17:16:42		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
build	18-Feb-2019 17:16:42		at java.lang.reflect.Method.invoke(Method.java:498)
build	18-Feb-2019 17:16:42		at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
build	18-Feb-2019 17:16:42		at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
build	18-Feb-2019 17:16:42		at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
build	18-Feb-2019 17:16:42		at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
build	18-Feb-2019 17:16:42	Caused by: org.apache.maven.plugin.MojoExecutionException
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute(ScannerBootstrapper.java:67)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.maven.SonarQubeMojo.execute(SonarQubeMojo.java:104)
build	18-Feb-2019 17:16:42		at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
build	18-Feb-2019 17:16:42		at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
build	18-Feb-2019 17:16:42		... 19 more
build	18-Feb-2019 17:16:42	Caused by: java.lang.NullPointerException
build	18-Feb-2019 17:16:42		at org.sonar.dependencycheck.base.DependencyCheckUtils.cvssToSonarQubeSeverity(DependencyCheckUtils.java:43)
build	18-Feb-2019 17:16:42		at org.sonar.dependencycheck.DependencyCheckSensor.addIssue(DependencyCheckSensor.java:80)
build	18-Feb-2019 17:16:42		at org.sonar.dependencycheck.DependencyCheckSensor.addIssues(DependencyCheckSensor.java:189)
build	18-Feb-2019 17:16:42		at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:250)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:88)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:82)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:68)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:88)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:177)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:291)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:286)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:264)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
build	18-Feb-2019 17:16:42		at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
build	18-Feb-2019 17:16:42		at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
build	18-Feb-2019 17:16:42		at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
build	18-Feb-2019 17:16:42		at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:71)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
build	18-Feb-2019 17:16:42		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
build	18-Feb-2019 17:16:42		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
build	18-Feb-2019 17:16:42		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
build	18-Feb-2019 17:16:42		at java.lang.reflect.Method.invoke(Method.java:498)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
build	18-Feb-2019 17:16:42		at com.sun.proxy.$Proxy19.execute(Unknown Source)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:185)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:137)
build	18-Feb-2019 17:16:42		at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute(ScannerBootstrapper.java:65)
build	18-Feb-2019 17:16:42		... 22 more
build	18-Feb-2019 17:16:42	[ERROR] 
build	18-Feb-2019 17:16:42	[ERROR] 
build	18-Feb-2019 17:16:42	[ERROR] For more information about the errors and possible solutions, please read the following articles:
build	18-Feb-2019 17:16:42	[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
simple	18-Feb-2019 17:16:42	Failing task since return code of [/opt/apache-maven-3.2.2/bin/mvn -Djava.io.tmpdir=/opt/bamboo/temp/xxxx-xxx-JOB1 sonar:sonar -e -X -Dsonar.dependencyCheck.reportPath=target/dependency-check-report.xml -Dsonar.host.url=https://sonar.xxx.xx -Dsonar.login=** -Dsonar.password=** -Dsonar.branch=master] was 1 while expected 0
simple	18-Feb-2019 17:16:42	SONAR4BAMBOO: was not able to find a SonarQube result URL
```

**Comment by miksonx** [comment]
This answers the part of blank report from the widget link and under vulnerability all of OWASP-Dependency-Check are zeros. 

> The widget needs to be removed. SQ 6 no longer supports them. Clicking on 'Vulnerabilities' will display DC results and there will be a Dependency-Check option on the menu to display the DC HTML report (if you've specified it).

**Comment by Reamer** [comment]
@miksonx Why you are working with dependency-check-maven:aggregate? Can you switch to dependency-check-maven:check and test again?

**Comment by miksonx** [comment]
@Reamer proposed solution provide exact same error as using aggregate.
I already fixed my problem using org.owasp:dependency-check-maven:3.3.2:aggregate specifying version to use and report was visible in SQ under vulnerabilities.

I'm very curious on differences specifying check vs aggregate and version/na. So if you can please elaborate more.

P.S. changing the title of the error is not suitable when people try to google the exact problem.

**Comment by Reamer** [comment]
@miksonx Which version of org.owasp:dependency-check-maven did you use before?

We have an open issue for parsing the aggregate reports. See #15. I never tested the aggregate report. But I think this project should proof that parsing of aggregated reports works as good as normal reports. SonarQube 7.6 drops the concept of modules (https://www.sonarqube.org/sonarqube-7-6/). #106 

By the way, your title was too long and does not mention any really useful information except the NullPointerException. Please add error messages in issue text field. Google and other search engines should be able to parse also this information.

**Comment by miksonx** [comment]
@Reamer as described on the start there is no version specified. (org.owasp:dependency-check-maven:aggregate)
I didn't had time to review the class code of both modules but as already described on #15 what is from 2016 and was not able to google seems that describes the existing issue.
In my case SQ 6.7 LTS provided more compatibility.
I hope the #15  be resolved soon.

**Comment by s4s0l** [comment]
For me this happens with sonar Version 7.6 (build 21501) dependency check plugin in 1.2.1 and in gradle build im using:
```
"org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.7"
'org.owasp:dependency-check-gradle:5.0.0-M2' 
```

**Comment by Reamer** [comment]
At the moment we have no compatibility for sonarqube 7.6. See https://github.com/stevespringett/dependency-check-sonar-plugin/issues/106

**Comment by Reamer** [comment]
@s4s0l The parser doesn't support 'org.owasp:dependency-check-gradle:5.0.0-M2' take a look into #117.
Closing in favour of #117

**Comment by FelikZ** [comment]
@Reamer  Hey Philipp, I believe this issue is not solved yet.

I am using:
```
		<sonar-maven-plugin.version>3.6.1.1688</sonar-maven-plugin.version>
		<dependency-check-maven.version>5.2.2</dependency-check-maven.version>
```
Tried to update sonar-maven-plugin to 3.7 and got the same result every single run:

```
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.1.1688:sonar (default-cli) on project relay42-core-components: null: MojoExecutionException: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.1.1688:sonar (default-cli) on project relay42-core-components: null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:67)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at org.sonar.dependencycheck.base.DependencyCheckUtils.cvssToSonarQubeSeverity (DependencyCheckUtils.java:53)
    at org.sonar.dependencycheck.base.DependencyCheckUtils.cvssToSonarQubeSeverity (DependencyCheckUtils.java:71)
    at org.sonar.dependencycheck.reason.DependencyReason.addIssue (DependencyReason.java:92)
    at org.sonar.dependencycheck.reason.DependencyReasonSearcher.linkIssuesToDependencyReasons (DependencyReasonSearcher.java:168)
    at org.sonar.dependencycheck.reason.DependencyReasonSearcher.addDependenciesToInputComponents (DependencyReasonSearcher.java:125)
    at org.sonar.dependencycheck.DependencyCheckSensor.execute (DependencyCheckSensor.java:91)
    at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse (AbstractSensorWrapper.java:48)
    at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute (ProjectSensorsExecutor.java:49)
    at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart (ProjectScanContainer.java:363)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart (GlobalContainer.java:126)
    at org.sonar.core.platform.ComponentContainer.startComponents (ComponentContainer.java:136)
    at org.sonar.core.platform.ComponentContainer.execute (ComponentContainer.java:122)
    at org.sonar.batch.bootstrapper.Batch.doExecute (Batch.java:73)
    at org.sonar.batch.bootstrapper.Batch.execute (Batch.java:67)
    at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute (BatchIsolatedLauncher.java:46)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke (IsolatedLauncherProxy.java:60)
    at com.sun.proxy.$Proxy27.execute (Unknown Source)
    at org.sonarsource.scanner.api.EmbeddedScanner.doExecute (EmbeddedScanner.java:185)
    at org.sonarsource.scanner.api.EmbeddedScanner.execute (EmbeddedScanner.java:137)
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:65)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Version 4.0.2 was working perfectly fine until recently XML feed was deprecated:
https://nvd.nist.gov/General/News/XML-Vulnerability-Feed-Retirement-Phase-3

And we have to switch to a later version of dependency-check-plugin to use json feeds.

Any idea how to workaround NullPointerException? There is no way to rollback anymore.

Thank in advance

**Comment by Reamer** [comment]
Hi @FelikZ,
which version of dependency-check-sonar-plugin do you use? Recommended version is [1.2.6](https://github.com/dependency-check/dependency-check-sonar-plugin/releases/tag/1.2.6).

**Comment by FelikZ** [comment]
@Reamer we are using 1.2.2. Could 1.2.6 make a difference? I think it fails before it reaches SQ but I might be totally wrong.

**Comment by Reamer** [comment]
Hi @FelikZ,
1.2.6 can make a difference. The sonar-maven-plugin creates a connection to Sonarqube and executes parts of the dependency-check-sonar-plugin on your local machine.

**Comment by Reamer** [comment]
@FelikZ Please let me know, if your problem is solved.

**Comment by FelikZ** [comment]
@Reamer I have just finished testing it and yes, I can confirm that combination of:
```
		<sonar-maven-plugin.version>3.6.0.1398</sonar-maven-plugin.version>
		<dependency-check-maven.version>5.2.2</dependency-check-maven.version>
```
and
```
dependency-check-sonar-plugin 1.2.6
```
Fixes #107 

Thank you!
